src/EventTickets/Routes/GetEventFormsListTable.php

```diff
@@ -0,0 +1,212 @@
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getEventForms(): array
+    {
+        $page = $this->request->get_param('page');
+        $perPage = $this->request->get_param('perPage');
+        $sortColumns = ['id'];
+        $sortDirection = $this->request->get_param('sortDirection') ?: 'asc';
+
+        $query = DonationForm::query();
+        $query = $this->getWhereConditions($query);
+
+        foreach ($sortColumns as $sortColumn) {
+            $query->orderBy($sortColumn, $sortDirection);
+        }
+
+        $query->limit($perPage)
+            ->offset(($page - 1) * $perPage);
+
+        $forms = $query->getAll();
+
+        if (!$forms) {
+            return [];
+        }
+
+        return $forms;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getEventFormsCount(): int
+    {
+        $query = DonationForm::query();
+        $query = $this->getWhereConditions($query);
+
+        return $query->count();
+    }
+
+    /**
+     * @unreleased
+     */
+    private function getWhereConditions(QueryBuilder $query): QueryBuilder
+    {
+        $eventIdPattern = sprintf('"eventId":%s', $this->request->get_param('event_id'));
+
+        $query->whereLike('give_formmeta_attach_meta_fields.meta_value', '%"name":"givewp/event-tickets"%')
+            ->where(function ($query) use ($eventIdPattern) {
+                $query->whereLike(
+                    'give_formmeta_attach_meta_fields.meta_value',
+                    "%$eventIdPattern}%"
+                ) // When the eventId is the only block attribute.
+                ->orWhereLike(
+                    'give_formmeta_attach_meta_fields.meta_value',
+                    "%$eventIdPattern,%"
+                ); // When the eventId is the NOT only block attribute.
+            });
+
+        return $query;
+    }
+
+
+    /**
+     * @unreleased
+     *
+     * @return bool|\WP_Error
+     */
+    public function permissionsCheck()
+    {
+        return current_user_can('edit_posts')?: new \WP_Error(
+            'rest_forbidden',
+            esc_html__("You don't have permission to view Events", 'give'),
+            ['status' => is_user_logged_in() ? 403 : 401]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    private function prepareItems(array $forms): array
+    {
+        return array_map(
+            function (DonationForm $form) {
+                return $this->formatColumns($form);
+            },
+            $forms
+        );
+    }
+
+    private function formatColumns(DonationForm $form): array
+    {
+        $formEditLink = get_edit_post_link($form->id);
+        $donationIds = give()->donations->getAllDonationIdsByFormId($form->id) ?? [];
+        $soldTicketsCount = count($donationIds) > 0
+            ? EventTicket::query()
+                ->where('event_id', $this->request->get_param('event_id'))
+                ->whereIn('donation_id', $donationIds)
```

Comment:
This could probably be combined into a single query with a `join` instead of a `whereIn`.